ironfish/src/rpc/routes/wallet/createAccount.ts

```diff
@@ -46,23 +45,24 @@ routes.register<typeof CreateAccountRequestSchema, CreateAccountResponse>(
   async (request, context): Promise<void> => {
     AssertHasRpcContext(request, context, 'wallet')
 
-    let account
-    try {
-      account = await context.wallet.createAccount(request.data.name)
-    } catch (e) {
-      if (e instanceof DuplicateAccountNameError) {
-        throw new RpcValidationError(e.message, 400, RPC_ERROR_CODES.DUPLICATE_ACCOUNT_NAME)
-      }
-      throw e
+    const name = request.data.name
+
+    if (context.wallet.accountExists(name)) {
+      throw new RpcValidationError(
+        `There is already an account with the name ${name}`,
+        400,
+        RPC_ERROR_CODES.ACCOUNT_EXISTS,
+      )
```

Comment:
Was this change intentional? This introduces a concurrency problem (two concurrent calls with the same `name` may both pass the `accountExists` test and then both call `createAccount`, resulting in one call overriding the other without giving any hint of what occurred to the user)

Reply:
unintentional, reverting it. merge got messed up somehow